kinto/views/admin.py

```diff
@@ -0,0 +1,134 @@
+"""
+Special views for administration.
+"""
+
+import itertools
+import colander
+
+from kinto.authorization import RouteFactory
+from kinto.core import utils as core_utils, resource
+from kinto.core.events import notify_resource_event, ACTIONS
+from kinto.core.resource import viewset
+from kinto.core.storage import Filter, Sort
+from kinto.core.storage.utils import paginated
+
+
+def slice_into_batches(iterable, batch_size):
+    # Taken from https://code.activestate.com/recipes/303279-getting-items-in-batches/
+    i = iter(iterable)
+    while True:
+        batchiter = itertools.islice(i, batch_size)
+        yield itertools.chain([next(batchiter)], batchiter)
+
+
+class Deleted(resource.ResourceSchema):
+    principal = colander.SchemaNode(colander.String())
+
+    class Options:
+        preserve_unknown = False
+
+
+class UserDataFactory(RouteFactory):
+    def _find_required_permission(self, request, service):
+        return ('user_data', 'delete')
```

Comment:
I did this so I could control the permission name I wanted to show up in the config. (I wanted `user-data_delete` instead of `user-data_write`.) I just noticed this one says `user_data` instead of `user-data`, even though my config file says `user-data`, so I guess I don't 100% understand how this works. There might also be a better way to get this to behave the way I want to, rather than overriding the factory.